### PR TITLE
Add context for the All translatable string and enforce l10n best practices

### LINKED ISSUE #58185: Review all the 'All' translatable strings and add context
### Description

Noticed while working on https://github.com/WordPress/gutenberg/pull/58117

Currently, there's a total of 8 occurrences in the codebase of a `__( 'All' )` translatable string.

While this is okay in English, it is not for other languages who have genres. As required by the WordPress localization standards, these cases must be disambiguated by adding context, to allow translators to traduce the string correctly.

For example, in Italian `All` can be both masculine or feminine depending on the subject it refers to:
- In Italian 'Patterns` is maschile e.g. 'Tutti i pattern'. In this case `All' must be translated to 'Tutti`.
- In Italian 'Categories` is feminine e.g. 'Tutte le categorie' and `All` must be translated to 'Tutte'.

All these cases, and similar ones, must use the `_x()` funciton instead of `__()` to specify a context.

### Step-by-step reproduction instructions

Search iin the codebase for all occurrences of `__( 'All' )`.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by afercia**
Fixes https://github.com/WordPress/gutenberg/issues/58185

## What?

The `__( 'All'  )` string needs more context to be correctly translatable.

## Why?

Languages other htan English may have genres.

## How?

- Add context via the `_x()` function.
- Fixes usage of the `_x()` second parameter: must be short and lowercase.
- Removes unnecessary translators comments.
- Improves existing translators comments.

## Testing Instructions

Not fully appplicable.
I guess a code review is enough in this case.

### CODE REVIEW

**Comment by afercia** [comment]
I think I accidentally included an unrelated fix in packages/edit-site/src/components/global-styles/font-library-modal/library-font-card.js but I guess it's not a big deal, it's a valid fix.

File: packages/components/src/query-controls/index.tsx [hunk 1]
```diff
@@ -96,12 +96,12 @@ export function QueryControls( {
 								value: 'date/asc',
 							},
 							{
-								/* translators: label for ordering posts by title in ascending order */
+								/* translators: Label for ordering posts by title in ascending order.translators: */
 								label: __( 'A → Z' ),
```
Comment (carolinan) [inline packages/components/src/query-controls/index.tsx:100]: I believe there is an accidental typo here where `translators:` is added at the end of the comment.

Comment (afercia) [reply packages/components/src/query-controls/index.tsx:100]: Oh yes thank you

**Comment by afercia** [comment]
I'd appreciate a quick review, changes in this PR are simple enough Cc @WordPress/gutenberg-core 
I don't think the components CHANGELOG needs a new entry for a string change.

**Review by Mamaduka — APPROVED** [review]
The changes look good to me.

**Comment by afercia** [comment]
Thanks @Mamaduka 👋

**Pull request merged**